aws_utils/s3/s3_utils.py

```diff
@@ -487,3 +488,75 @@ def rename_keys_on_s3(bucket_name, bucket_region, prefix_root, prefix_modificati
                 rename_s3_key(boto3_client, bucket_name, current_prefix, new_prefix_name)
             except (ClientError, AttributeError) as e:
                 logger.error('Unable to rename key prefix {}, {}'.format(current_prefix, e[0]))
+
+
+def get_s3_keys_by_regex(s3_bucket, s3_directory, pattern):
+    """Fetches the s3 keys of all files within the supplied directory using a regex
+
+    Args:
+        s3_bucket (s3 bucket obj):
+        s3_directory (str):
+        pattern (regex pattern obj)
+    Returns: ([boto s3 key, boto s3 key...]):
+    Raises: ValueError
+    """
+    bucket_contents = s3_bucket.list(s3_directory)
+    s3_filepaths = [key for key in bucket_contents if pattern.search(key.name)]
+    if not s3_filepaths:
+        raise ValueError('No valid segment files found in {}'.format(s3_directory))
```

Comment:
It can be tricky to decide when something should throw an error vs return a null result. The question you should answer is how suprised would I be to get the result, do I ever want to keep running without knowing it happened. Here I would err on the side of returning an empty list is fine